modxcms/revolution
# Fix Role Authority Editability

### What does it do?
Add new listener routine to restrict editing based on whether a Role is assigned to an ACL entry. Also:

1. Adds new UI feedback, both a lock icon (instead of pencil) on hover of authority as well as an alert dialog when double-clicking on locked authority
2. Ensures authority cannot be deleted in grid editor
3. Remove pencil icon for cells of non-editable Roles

### Why is it needed?
The ability to edit an assigned Role's authority leads to orphaned ACL rules that not longer show up in the manager, yet remain in the database. See the referenced issue below.

### How to test

1. Rebuild template (grunt build) and clear manager and browser caches
2. Create a few Roles under the Access Control Lists section
3. Assign at least one Role to any ACL entry of your choice
4. Verify that the assigned Roles' authority is locked in the Roles grid
5. Verify that unassigned Roles remain fully editable in the Roles grid (authority is unlocked)

### Note
The initial commit contains all substantive changes, while the follow up is just code-style/optimization oriented.

### Related issue(s)/PR(s)
Resolves #16565
